### PR TITLE
Support 'Compute' for MeshImporter.color

### PR DESCRIPTION
**Pull request by TimsLuxo**
MeshImporter.color=Compute forces constant to vertex and uses first color as vertex color if other methods fail

### CODE REVIEW

**Review by jcowles — APPROVED** [review]

**Comment by jcowles** [comment]
Holding this to merge until after preview1 has been released

**Comment by TimsLuxo** [comment]
> Holding this to merge until after preview1 has been released

If possible, please merge this, as we have several use cases that currently require fragile workarounds. Thanks.

**Comment by jcowles** [comment]
Doh, sorry to leave this hanging so long!

**Comment by jcowles** [comment]
Hey @TimsLuxo, this new compute code path only gets hit when the mesh colors are broken or are uniform, right?

Are you actually just trying to add support for uniform colors?

File: package/com.unity.formats.usd/Runtime/Scripts/IO/Geometry/MeshImporter.cs [hunk 1]
```diff
@@ -506,13 +512,29 @@ public class GeometrySubsets {
...
                 new Exception("Error unrolling Face-Varying colors at <" + path + ">", ex));
           }
         } else {
           Debug.LogWarning("Uniform (color per face) display color not supported");
         }
+
+        if (forceVertexColors) {
+            Color [] colors = new Color[usdMesh.points.Length];
+            if (usdMesh.colors.Length != usdMesh.points.Length) {
+              Color color = usdMesh.colors[0].gamma;
```
Comment (jcowles) [inline package/com.unity.formats.usd/Runtime/Scripts/IO/Geometry/MeshImporter.cs]: Do you actually want `gamma` here?

I think we only want the color in gamma space when it's used as a material parameter, and now I'm questioning if that's even correct for HDRP/UniversalRP...

Comment (TimsLuxo) [reply package/com.unity.formats.usd/Runtime/Scripts/IO/Geometry/MeshImporter.cs]: Good point. I suppose not in this case. Likewise with *RP which are linear, correct?

Comment (jcowles) [reply package/com.unity.formats.usd/Runtime/Scripts/IO/Geometry/MeshImporter.cs]: Yeah, I just confirmed the SRP pipelines are linear, so we have a bug in the other code calling .gamma above.

**Review by jcowles — CHANGES_REQUESTED** [review]
Would be nice to kill the .gamma call (ultimate fix TBD) and perhaps refactor this to just be the default when we find uniform colors?

**Comment by TimsLuxo** [comment]
> Would be nice to kill the .gamma call (ultimate fix TBD) and perhaps refactor this to just be the default when we find uniform colors?

This is implemented in latest update to pull request.

**Comment by jcowles** [comment]
I'm sincerely sorry for sitting on this for so long @TimsLuxo - I'm going to close this PR, since #261 refactors so much of the import code.

If you ever want to revisit this, ping me and I'll make it happen.

**Pull request closed without merging**